src/pip/_internal/utils/misc.py

```diff
@@ -1182,8 +1183,8 @@ def protect_pip_from_modification_on_windows(modifying_pip):
     """
     pip_names = [
         "pip.exe",
-        "pip{}.exe".format(sys.version_info[0]),
-        "pip{}.{}.exe".format(*sys.version_info[:2])
+        "pip{}.exe".format(*sys.version_info),
+        "pip{}.{}.exe".format(*sys.version_info),
```

Comment:
Like here

Reply:
This wasn't in your patch, I accidentally amended the wrong commit though, let me know if you'd like me to reattribute though I suspect it will get squashed anyway

Reply:
Okay, well, I was going off your comment that you just did a rebase, but it looks like you introduced new changes in the process. Can you incorporate the helper function into the new changes?

Reply:
I was going to, but I think it makes it worse /  less readable here as it can only be applied to one of the cases and not both

which goes back to my original point in that I think the helper is a bad idea

Reply:
I just took this out instead